Document Pod affinity namespaceSelector (#26820)
kubernetes/website: content/en/docs/concepts/scheduling-eviction/assign-pod-node.md

All `matchExpressions` associated with `requiredDuringSchedulingIgnoredDuringExecution` affinity and anti-affinity
must be satisfied for the pod to be scheduled onto a node.

> #### Namespace Selector

Reviewer: Style nit:
```suggestion
#### Namespace selector
```

Author: done.

{{< feature-state for_k8s_version="v1.21" state="alpha" >}}

> Users can also select namespaces using `namespaceSelector`, which is a label query over the set of namespaces.

Reviewer 2: nit: `matching namespaces`

The affinity term is applied to the union of the namespaces selected by `namespaceSelector` and the ones listed in the `namespaces` field.

> Note that an empty `namespaceSelector` ({}) matches all namespaces.

Reviewer 2: if omitted or null, as well as the `namespaces` list, affinity/anti-affinity terms only match the namespace of the pod that defines them.

Author: updated

> This feature is alpha and disabled by default. You can enbale it by setting the

Reviewer 2: enable

> This feature is currently alpha and disabled by default, and can be enabled via 

Reviewer: Style nit:
```suggestion
This feature is alpha and disabled by default. You can enable the it by setting the
```

Author: done.

[feature gate](/docs/reference/command-line-tools-reference/feature-gates/)

> `PodAffinityNamespaceSelector` in both kube-apiserver and kube-scheduler.

Reviewer: Please add this feature gate into https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/ and provide a brief description.

Author: updated.

#### Namespace selector
{{< feature-state for_k8s_version="v1.21" state="alpha" >}}

Users can also select matching namespaces using `namespaceSelector`, which is a label query over the set of namespaces.
The affinity term is applied to the union of the namespaces selected by `namespaceSelector` and the ones listed in the `namespaces` field.
Note that an empty `namespaceSelector` ({}) matches all namespaces, while a null or empty `namespaces` list and 

> null `namespaceSelector` means "this pod's namespace"

Reviewer 2: end sentence with period.
